src/main/scala/chiseltest/legacy/backends/vcs/VcsExecutive.scala

```diff
@@ -122,7 +122,7 @@ object VcsExecutive extends BackendExecutive {
       .collectFirst[Seq[String]] {
         case TestCommandOverride(f) => f.split(" +")
       }
-      .getOrElse { Seq(new File(targetDir, circuit.name).toString) }
+      .getOrElse { Seq(new File(targetDir, circuit.name).toString) } ++ coverageFlags
```

Comment:
`coverageFlags` is part of `vcsFlags` on line 108, why does it need to be duplicated here? Why does this need to be handled differently from the other flags?

Reply:
Yeah, the duplication is needed due to how coverage works in the VCS flow. The coverage flags passed into the vcs command on line 108 tells the tool to compile the design for the types of coverage. The same flags are also needed in the simulation command to tell the simulator which coverage to actively monitor.

This seems kind of redundant, but it allows for designs to be compiled for several types of coverage while the simulator can monitor them in isolation (e.g. while the design is compiled with line + branch coverage, a specific test can only monitor for line coverage).

Reply:
I found this piece of documentation helpful: http://read.pudn.com/downloads343/ebook/1502221/cov_ug.pdf (page 47, 2-35)

Reply:
Makes sense. Can you add an abbreviated version of that explanation in the comments so it's obvious to others looking at the code why it's being duplicated?

Also, the book link doesn't work (anymore?) 🙃.

I think that's the only change I'd like.

Reply:
Sounds good. Just committed!